api/src/paths/taxonomy/taxon/tsn/hierarchy.ts

```diff
@@ -0,0 +1,106 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { getAPIUserDBConnection } from '../../../../database/db';
+import { ItisService } from '../../../../services/itis-service';
+import { getLogger } from '../../../../utils/logger';
+
+const defaultLog = getLogger('paths/taxonomy/taxon/tsn/hierarchy');
+
+export const GET: Operation = [getTaxonHierarchyByTSN()];
+
+GET.apiDoc = {
+  description: 'Get taxon records by TSN ids.',
+  tags: ['taxon_id'],
+  security: [
+    {
+      Bearer: []
+    }
+  ],
+  parameters: [
+    {
+      description: 'Taxon TSN ids.',
+      in: 'query',
+      name: 'tsn',
+      schema: {
+        type: 'array',
+        description: 'One or more Taxon TSN ids.',
+        items: {
+          type: 'integer',
+          minimum: 0,
+          minItems: 1,
+          maxItems: 100
+        }
+      },
+      required: true
+    }
+  ],
+  responses: {
+    200: {
+      description: 'Taxonomy response.',
+      content: {
+        'application/json': {
+          schema: {
+            type: 'array',
+            items: {
+              title: 'Species',
+              type: 'object',
+              required: ['tsn', 'hierarchy'],
+              properties: {
+                tsn: {
+                  type: 'integer'
+                },
+                hierarchy: { type: 'array', items: { type: 'integer' } }
+              },
+              additionalProperties: false
+            }
+          }
+        }
+      }
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+/**
+ * Get taxon by ITIS TSN.
+ *
+ * @returns {RequestHandler}
+ */
+export function getTaxonHierarchyByTSN(): RequestHandler {
+  return async (req, res) => {
+    defaultLog.debug({ label: 'getTaxonHierarchyByTSN', message: 'query params', query: req.query });
+
+    const connection = getAPIUserDBConnection();
+
+    const tsnIds: number[] = (req.query.tsn as (string | number)[]).map(Number);
```

Comment:
super nit: 
Regarding `as (string | number)[]`:
The query params will always be a string array at this stage. All http params that aren't in the body get converted to strings.